src/control/server/ctl_storage_rpc.go

```diff
@@ -151,6 +165,102 @@ func (c *ControlService) scanScm(ctx context.Context, req *ctlpb.ScanScmReq) (*c
 	return newScanScmResp(c.getScmUsage(ssr))
 }
 
+// Adjust the NVME available size to its real usable size.
+func (c *ControlService) adjustNvmeSize(resp *ctlpb.ScanNvmeResp) {
+	for _, ctl := range resp.GetCtrlrs() {
+		for _, dev := range ctl.GetSmdDevices() {
+			if dev.GetDevState() != "NORMAL" {
+				c.log.Infof("WARNING: Adjusting available size of unusable SMD "+
+					"device %s (ctlr %s) to O Bytes: device state %q",
+					dev.GetUuid(), ctl.GetPciAddr(), dev.GetDevState())
+				dev.AvailBytes = 0
+				continue
+			}
+			if dev.GetClusterSize() == 0 || len(dev.GetTgtIds()) == 0 {
+				c.log.Errorf("Skipping device %s (%s) with missing storage info",
+					dev.GetUuid(), ctl.GetPciAddr())
+				continue
+			}
+
+			targetCount := uint64(len(dev.GetTgtIds()))
+			unalignedBytes := dev.GetAvailBytes() % (targetCount * dev.GetClusterSize())
+			c.log.Debugf("Adjusting available size of SMD device %s (ctlr %s): "+
+				"excluding %s (%d Bytes) of unaligned storage",
+				dev.GetUuid(), ctl.GetPciAddr(),
+				humanize.Bytes(unalignedBytes), unalignedBytes)
+			dev.AvailBytes -= unalignedBytes
+		}
+	}
+}
+
+// return the size of the ram disk file used for managing SCM metadata
+func (c *ControlService) getMetadataCapacity(mountPoint string) (uint64, error) {
+	var engineCfg *engine.Config
+loop:
```

Comment:
nit: ugh, :-) always try to avoid these personally but there's no rules against them

Reply:
- [ ] Remove labeled break

Reply:
- [ ]  Fix SMD device status test